benchmark_runner/benchmark_operator/benchmark_operator_workloads.py

```diff
@@ -422,30 +548,32 @@ def uperf_pod(self, name: str=''):
                 self.__oc.wait_for_initialized(label=label, workload=workload, label_uuid=False)
                 self.__oc.wait_for_ready(label=label, workload=workload, label_uuid=False)
             # uperf client
-            self.__oc.wait_for_pod_create(pod_name=f'uperf-client')
+            self.__oc.wait_for_pod_create(pod_name='uperf-client')
             self.__oc.wait_for_initialized(label='app=uperf-bench-client', workload=workload)
             self.__oc.wait_for_ready(label='app=uperf-bench-client', workload=workload)
-            self.__oc.wait_for_pod_completed(label='app=uperf-bench-client', workload=workload)
+            status = self.__oc.wait_for_pod_completed(label='app=uperf-bench-client', workload=workload)
+            status = 'complete' if status else 'failed'
             # system metrics
             if environment_variables.environment_variables_dict['system_metrics'] == 'True':
                 self.system_metrics_collector(workload=workload)
+            # save run artifacts logs
+            run_artifacts_url = self.__create_run_artifacts(workload=workload, labels=['uperf-client', 'uperf-server'])
```

Comment:
I'd really like this to be hoisted from the individual workload functions into `run_workload_func`.  It will make adding the metrics much easier and save a lot of code.

Reply:
Upon discussion, this can be done by wrapping `run_workload_func` so there is no issue here.